pkg/controller/eventing/subscription/reconcile_test.go

```diff
@@ -775,6 +803,64 @@ func TestAllCases(t *testing.T) {
 	}
 }
 
+type finalizerTestCase struct {
```

Comment:
My preference is defining this in line in the test.

```golang
testcases := []struct {
	name     string
	original sets.String
	add      bool
	want     sets.String
}{
    {
        name: ...
```

Reply:
+1 unless you intend to reuse the test cases object.